app/controllers/support_interface/application_forms/application_choices_controller.rb

```diff
@@ -0,0 +1,40 @@
+module SupportInterface
+  module ApplicationForms
+    class ApplicationChoicesController < SupportInterfaceController
+      before_action :build_application_form, :build_application_choice, :redirect_to_application_form_unless_declined_and_flag_active
+
+      def reinstate_offer
```

Comment:
I mentioned this before, but this really should be `confirm_reinstate_offer` as that's what the user is doing on the page

The patch endpoint  is what is actually doing the work of reinstating the offer